### ISSUE TITLE
faas-cli 0.10.1 brew upgrade issue

### ISSUE THREAD

**Issue by chenrui333** [openfaas/faas-cli#720]
Relates to https://github.com/Homebrew/homebrew-core/pull/46345

Context are in jenkins builds
https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/51263/version=mojave/console

The above is Mojave build log.

**Comment by alexellis** [comment]
@johnmccabe @rgee0 @martindekov PTAL?

**Comment by alexellis** [comment]
@martindekov this could be regression after #668 ?

```
$ faas-cli template pull
The repository URL must be a valid git repo uri
```

**Comment by alexellis** [comment]
Prior version:

```
[alex@nuc ~]$ faas-cli version
  ___                   _____           ____
 / _ \ _ __   ___ _ __ |  ___|_ _  __ _/ ___|
| | | | '_ \ / _ \ '_ \| |_ / _` |/ _` \___ \
| |_| | |_) |  __/ | | |  _| (_| | (_| |___) |
 \___/| .__/ \___|_| |_|_|  \__,_|\__,_|____/
      |_|

CLI:
 commit:  0e1c4bb76df42607e7435ba1110dfbdcfc133f95
 version: 0.8.18

Gateway
 uri:     http://127.0.0.1:8080
 version: dev
 sha:     777a1872621b99b243cd06a063a3a59a8023b43b
 commit:  Scale functions with namespace option


Provider
 name:          faas-swarm
 orchestration: swarm
 version:       0.7.2 
 sha:           ef7d9132e29edc8ff3962685ba772ad6e3f01bda
[alex@nuc ~]$ faas-cli template pull
Fetch templates from repository: https://github.com/openfaas/templates.git at master
2019/11/07 16:18:03 Attempting to expand templates from https://github.com/openfaas/templates.git

2019/11/07 16:18:04 Cannot overwrite the following 11 template(s): [csharp dockerfile go go-armhf node node-arm64 node-armhf python python-armhf python3 ruby]
2019/11/07 16:18:04 Fetched 7 template(s) : [csharp-armhf dockerfile-armhf java12 java8 node12 php7 python3-armhf] from https://github.com/openfaas/templates.git
```

**Comment by rgee0** [comment]
I think this might be implicated:

https://github.com/openfaas/faas-cli/pull/668/files#diff-765f4ba59499cd6d3963f583eb6012b1L47

Not sure what this part does if there isn't an arg to set `repository`:

https://github.com/openfaas/faas-cli/pull/668/files#diff-765f4ba59499cd6d3963f583eb6012b1R46

**Comment by martindekov** [comment]
Will have a better look in an hour ^

**Comment by rgee0** [comment]
Could do with some independent testing: https://github.com/openfaas/faas-cli/pull/721

**Comment by chenrui333** [comment]
I think I still have issue on testing the formula on the latest release

```
Still failed locally (Catalina OS)
$ brew test faas-cli
Testing faas-cli
==> /usr/local/Cellar/faas-cli/0.10.3/bin/faas-cli deploy -yaml test.yml 2>&1
==> /usr/local/Cellar/faas-cli/0.10.3/bin/faas-cli template pull 2>&1
==> /usr/local/Cellar/faas-cli/0.10.3/bin/faas-cli new --list
==> /usr/local/Cellar/faas-cli/0.10.3/bin/faas-cli deploy -yaml test.yml
Error: faas-cli: failed
An exception occurred within a child process:
Test::Unit::AssertionFailedError: <0> expected but was
<1>.
```
https://github.com/Homebrew/homebrew-core/pull/46345#issuecomment-551336131

**Comment by rgee0** [comment]
@chenrui333 can you shed some light on why the first deploy command doesn't fail but the second does?  It looks as though the test is trying to match some output that we wont actually see.  Trouble is, I cant understand why this behaviour would have changed - it should have been failing for some time.

Also, the `test.yml` file contains disallowed characters:
```
./faas-cli-darwin new --lang=python dummy_function
Function name can only contain a-z, 0-9 and dashes
```

**Comment by martindekov** [comment]
I see mojave failed https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/51428/version=mojave/console

Also seeing comment on last release `0.9.5` that mojave is failing:
https://github.com/Homebrew/homebrew-core/pull/45596

Could this be the reason? For reference here is the ruby script:

https://github.com/Homebrew/homebrew-core/blob/master/Formula/faas-cli.rb

@rgee0 I don't see `faas-cli new` I see a string written directly into a file. `new` is used only once `faas-cli new --list`

**Comment by rgee0** [comment]
@martindekov perhaps there's an easier way to demonstrate that the hand-crafted file uses characters that are unsupported whilst also conveying which characters are supported?

**Comment by martindekov** [comment]
Hey Richard, I don't disregard your comment, it is very valid, and I believe should be reflected in that file. Just trying to point that the current situation(possibly exact test failure) happened before in the same environment.

**Comment by chenrui333** [comment]
@martindekov

> I see mojave failed https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/51428/version=mojave/console

Jenkins build is on PR basis, and a good place to go for the reference. 

> Also seeing comment on last release 0.9.5 that mojave is failing:
Homebrew/homebrew-core#45596

I think that is a CI hiccup, all green after the rebuild.

**Comment by chenrui333** [comment]
Looks like the formula test got fixed in 0.10.5. Closing the issue now.
